File: python/2.7/factory/pizzaIngredientFactory/ChicagoPizzaIngredientFactory.py
import sys

from PizzaIngredientFactory import *

sys.path.append("./dough")
from ThickCrustDough import *

sys.path.append("./sauce")
from PlumTomatoSauce import *

sys.path.append("./cheese")
from MozzarellaCheese import *

sys.path.append("./veggies")
from BlackOlives import *
from Spinach import *
from Eggplant import *

sys.path.append("./pepperoni")
from SlicedPepperoni import *

sys.path.append("./clams")
from FrozenClams import *

class ChicagoPizzaIngredientFactory(PizzaIngredientFactory):
    def __init__(self):
        super(ChicagoPizzaIngredientFactory, self).__init__()

    def createDough(self):
        return str(ThickCrustDough())

    def createSauce(self):
        return str(PlumTomatoSauce())

    def createCheese(self):
        return str(MozzarellaCheese())

    def createVeggies(self):
        self.veggies = [str(BlackOlives()), str(Spinach()), str(Eggplant())]
        return self.veggies

    def createPepperoni(self):
        return str(SlicedPepperoni())

    def createClam(self):
        return str(FrozenClams())

File: java/1.7/factory/src/com/pizzaIngredientFactory/ChicagoPizzaIngredientFactory.java
package com.pizzaIngredientFactory;

import com.cheese.Cheese;
import com.cheese.MozzarellaCheese;
import com.clams.Clams;
import com.clams.FrozenClams;
import com.dough.Dough;
import com.dough.ThickCrustDough;
import com.pepperoni.Pepperoni;
import com.pepperoni.SlicedPepperoni;
import com.sauce.PlumTomatoSauce;
import com.sauce.Sauce;
import com.veggies.BlackOlives;
import com.veggies.Eggplant;
import com.veggies.Spinach;
import com.veggies.Veggies;

public class ChicagoPizzaIngredientFactory 
	implements PizzaIngredientFactory 
{

	public Dough createDough() {
		return new ThickCrustDough();
	}

	public Sauce createSauce() {
		return new PlumTomatoSauce();
	}

	public Cheese createCheese() {
		return new MozzarellaCheese();
	}

	public Veggies[] createVeggies() {
		Veggies veggies[] = { new BlackOlives(), 
		                      new Spinach(), 
		                      new Eggplant() };
		return veggies;
	}

	public Pepperoni createPepperoni() {
		return new SlicedPepperoni();
	}

	public Clams createClam() {
		return new FrozenClams();
	}
}
